model/src/main/java/io/gravitee/definition/model/Api.java

```diff
@@ -118,22 +169,45 @@ public void setTags(Set<String> tags) {
     }
 
     public void setResources(List<Resource> resources) {
+        Set<String> distinct = new HashSet<>(resources.size());
+        resources.removeIf(p -> !distinct.add(p.getName()));
         this.resources = resources;
     }
 
+    @JsonIgnore
     public Map<String, Pattern> getPathMappings() {
         return pathMappings;
     }
 
+    @JsonIgnore
     public void setPathMappings(Map<String, Pattern> pathMappings) {
         this.pathMappings = pathMappings;
     }
 
-    public Map<String, ResponseTemplates> getResponseTemplates() {
+    @JsonGetter("path_mappings")
+    public Set<String> getPathMappingsJson() {
+        return pathMappings.keySet();
+    }
+
+    @JsonSetter("path_mappings")
+    public void setPathMappingsJson(Set<String> pathMappings) {
+        if (pathMappings == null) {
+            this.pathMappings = null;
+            return;
+        }
+        this.pathMappings = new LinkedHashMap<>();
+        pathMappings.forEach(pathMapping -> this.pathMappings.put(
+                pathMapping,
+                Pattern.compile(pathMapping.replaceAll(":\\w*", "[^\\/]*") + "/*")
+        ));
+    }
+
+    @JsonProperty("response_templates")
+    public Map<String, Map<String, ResponseTemplate>> getResponseTemplates() {
         return responseTemplates;
     }
 
-    public void setResponseTemplates(Map<String, ResponseTemplates> responseTemplates) {
+    public void setResponseTemplates(Map<String, Map<String, ResponseTemplate>> responseTemplates) {
         this.responseTemplates = responseTemplates;
     }
 
```

Comment:
For serialization of different versions we could also use [JsonViews](https://www.baeldung.com/jackson-json-view-annotation) to limit returned properties. But this would mean to set the right view each time the API get serialized. Please tell me your opinion about that.